### ISSUE TITLE
Agent not receiving notifications

### ISSUE THREAD

**Issue by fredcido** [LivePersonInc/node-agent-sdk#51]
Hi there all,

I have been using the node-agent-sdk in order to implement a chatbot, however, there might be a misunderstanding from my side. 

When using the simple [**greeting**](https://github.com/LivePersonInc/node-agent-sdk/tree/master/examples/greeting-bot) example, the agent starts to listen nicely, but after connecting as customer, both using https://livepersoninc.github.io/visitor-page/?siteid=__SITEID__ and https://codepen.io/liveperson/pen/xRzXXd?editors=0010 test pages, the agent doesn't seem to receive any notification that a new ring/customer has entered and is waiting to connect.

Only after sending some message from the customer side the bot then receives a notification and can pick up the conversation. I have tested all the examples provided in here, but none of them has the desired behavior.

When using the Chat API, https://github.com/LivePersonInc/agent-sample-app, it works as expected, the agent is able to pick up the incoming chats.

**Comment by donmanguno** [comment]
Hi @fredcido.

Chat and Messaging work differently.

In Chat, as soon as a visitor clicks a chat button (or after they submit the subsequent survey if enabled) their chat will begin and will be presented to eligible agents.

In Messaging, it is only after the consumer sends their first message that their conversation is presented to eligible agents.

**Comment by donmanguno** [comment]
@fredcido did this answer your question?

**Comment by fredcido** [comment]
@donmanguno Yes, it does. But it would be nice if you could recommend me then the best approach for the bot to pick up incoming chats/customers. Is there any recommended best practices to implement Chat and Messaging  together in order to deliver a better user experience?

**Comment by donmanguno** [comment]
Hi @fredcido,

In our system Chat and Messaging are entirely separate channels. We have a different [API for Chat Agent implementations](https://developers.liveperson.com/chat-agent-getting-started.html), and a basic [sample node.js application](https://github.com/LivePersonInc/agent-sample-app) implementing it.

As for best practice recommendations: how to best optimize your UX depends on the specific details of your site, your company, and the services you need to provide over text.  I am just a technical grunt and no expert in these questions, but we have a lot of people here at LP whose job it is to help optimize your user experience across the products and channels we provide. If you have a LivePerson account representative I suggest speaking with them, and if you don't you can chat with our Sales team.

*Note that the sample application is merely a sample and provided without any guarantees.
